examples/hello-world/format-example.cpp

```diff
@@ -0,0 +1,130 @@
+
+#include "internalGroup1.h"
+#include "slang.h"
+#include <algorithm>
+
+#include "includeGroup2.h"
+#include <vector>
+
+#include <vulkan.h>
+
+#ifdef __linux__
+#include <X11/Xlib.h>
+#else
+#include <Windows.h>
+#endif
+
+namespace slang
+{
+    /// <summary>
+    /// Summary of TestClass
+    /// </summary>
+    class TestClass
+    {
+    private:
+        int examplePrivateMember; // Trailing comment1
+        int* examplePointerMember = nullptr; // trailing coment2
+        float member4;
+        float member5;
+
+    public:
+        TestClass()
+            : examplePrivateMember(0), examplePointerMember(nullptr), member4(1.0f), member5(200.0f)
+        {
+        }
+
+        int shortFunction() { return 0; }
+
+        /// <summary>
+        /// A normal function.
+        /// </summary>
+        /// <param name="param1">The first parameter.</param>
+        /// <param name="param2">The second parameter</param>
+        /// <returns></returns>
+        int normalFunction(int param1, float* param2)
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                switch (i)
+                {
+                case 0:
+                {
+                    return normalFunction(0, nullptr);
+                }
+                break;
+                case 1:
+                {
+                    float b = *param2;
+                    i--;
+                }
+                break;
+                default:
+                    break;
+                }
+            }
+            if (param1)
+                return *param2;
+            else
+                return -1;
+        }
+        class LoweredValInfo
+        {
+        };
```

Comment:
I'd *strongly* prefer to be able to have more compact styles when we have empty `{}` blocks:

```c++
struct MyDerived : MyBase {}; // no need for more lines, please

virtual void handleSomeCase(...)
{} // nothing to do by default
```